frontend/src/components/ViewActivities/activitylist.js

```diff
@@ -2,18 +2,72 @@ import React from 'react';
 import * as activityFns from './activityfns.js';
 import ActivityDay from './activityday.js';
 import '../../styles/activities.css';
+import * as DB from '../../constants/database.js';
+import app from '../Firebase';
 
+const db = app.firestore();
+
+
+/**
+ * Gets the list of activities from the server. 
+ * 
+ * @param {string} tripId The trip ID.
+ */
+export async function getActivityList(tripId) {
+  return new Promise(function(resolve, reject) {
+    let tripActivities = [];
+    
+    db.collection(DB.COLLECTION_TRIPS).doc(tripId)
+    .collection(DB.COLLECTION_ACTIVITIES).get()
+    .then(querySnapshot => {
+      querySnapshot.forEach(doc => {
+        let data = doc.data();
+        data['id'] = doc.id;
+        
+        // TODO: if start date != end date, split into 2 days. (#37)
+
+        // Eliminate nanoseconds, convert to milliseconds.
+        data[DB.ACTIVITIES_START_TIME] =
+          data[DB.ACTIVITIES_START_TIME]['seconds'] * 1000;         
+        data[DB.ACTIVITIES_END_TIME] = 
+          data[DB.ACTIVITIES_END_TIME]['seconds'] * 1000;
+
+        tripActivities.push(data);
+      })
+    }).catch(error => {
+      console.log("It seems that an error has occured.");
+      tripActivities = null;
+    }).then( () => resolve(tripActivities) );
```

Comment:
I am definitely not all the way solid with Promises so I may be wrong here. But why do you need to return a new promise and then resolve it here? Cant you just return tripActivities from the initial .then()? Something like this 'should' return a Promise:

```
db.collection(DB.COLLECTION_TRIPS).doc(tripId)
    .collection(DB.COLLECTION_ACTIVITIES).get()
    .then(querySnapshot => {
      querySnapshot.forEach(doc => {
      ...
      })
     return tripActivities;

   })
   .catch(error => { ... }
   );
```

Reply:
The idea is to make the function finish before displaying, so that we don't end up displaying only half a trip/in the wrong order/something else. This is the result of  

Putting the query in a promise makes sure that you can't do anything with the returned data until the database call is completed, and resolving it at the end was only to make sure that we can access the tripActivities structure. This is how the query ran before we switched to React. 

Frankly, I didn't experiment with it after we switched to React.